File: llm_outputs/python_programs_corrected_by_mistral/possible_change.py
def possible_change(coins, total):
    if total == 0:
        return 1
    if total < 0:
        return 0

    first, *rest = sorted(coins) # changed 'first, *rest' to'sorted(coins)'
    return sum(possible_change(coins[:i+1], total-first) for i in range(len(coins))) + possible_change(rest, total)

File: llm_outputs/java_programs_corrected_by_llama/POSSIBLE_CHANGE.java
package java_programs;

import java.util.*;

public class PossibleChange {

    public static int possibleChange(int[] coins, int total) {
        if (coins.length == 0 || total <= 0) {
            return 0;
        } else if (total > 0 && total >= coins[0]) {
            return 1;
        } else {
            int first = coins[0];
            int[] rest = getRemainingCoins(coins, 1);

            return possibleChange(rest, total - first) +
                   possibleChange(getRemainingCoins(coins, 2), total);
        }
    }

    private static int[] getRemainingCoins(int[] coins, int start) {
        int length = coins.length - start;
        int[] result = new int[length];

        for (int i = 0; i < length; i++) {
            result[i] = coins[start + i];
        }

        return result;
    }
}